argmin/src/core/executor.rs

```diff
@@ -259,7 +259,15 @@ where
             }
         }
 
-        // in case it stopped prematurely and `termination_reason` is still `NotTerminated`,
+        if running.load(Ordering::SeqCst) {
+            // Solver execution has finished
+            running.store(false, Ordering::SeqCst);
+        } else {
+            // Solver execution has been interrupted manually
+            state = state.terminate_with(TerminationReason::KeyboardInterrupt);
+        }
```

Comment:
What is the reason for setting running to false if the solver has finished?

At first sight it seems to me that this should be sufficient:
```suggestion
        if !running.load(Ordering::SeqCst) {
            // Solver execution has been interrupted manually
            state = state.terminate_with(TerminationReason::KeyboardInterrupt);
        }
```

Reply:
Yep, it was my first version but somehow I was bothered to let the boolean called `running` to `true`, and yes I agree it feels awkward (even more now for there is no subsequent processing). Actually everything falls into place, if we use `interrupt` the inverse boolean of `running`.